src/worker/sw.ts

```diff
@@ -627,6 +627,44 @@ self.addEventListener('message', async (event) => {
       }
       break;
 
+    case 'PAUSE_SYNC':
+      // Handle pause as cancellation with manual pause flag
+      if (currentSyncService && syncInProgress) {
+        logInfo('sync', `Received PAUSE_SYNC message: ${message.reason || 'User requested pause'}`);
+        // Set manual pause flag in database
+        await DatabaseService.setManualPauseState(true);
+        // Cancel the sync
```

Comment:
This cleanup code is repeated a few times between the "PAUSE_SYNC", "CANCEL_SYNC", and various error handling blocks in this file. Unify the cleanup into a single function so that it's consistent.